tests/unit_tests/1001_lib_cpuset_validation_test.py

```diff
@@ -0,0 +1,117 @@
+import random
+import re
+
+
+from iocage_lib.ioc_json import IOCCpuset
+
+# For cpuset props we would like to test the following scenarios
+# 1) 0,1,2,3
+# 2) 0-3
+# 3) all
+# 4) off
+
+cpu_sets = IOCCpuset.retrieve_cpu_sets() + 1
```

Comment:
Unit test should never depend on execution environment. Also, you are restricting yourself to the number of CPUs test executor has. What if you need to test behavior with 1 CPU? With 128 CPUs? As we are testing only logic in `validate_cpuset_prop`, it is absolutely no important how many CPUs do we really have, only what `retrieve_cpu_sets` returns. You have to use `unittest.module` to fake it's return value in each test.